### ISSUE TITLE
MS Windows compilation painful

### ISSUE THREAD

**Issue by sebhz** [ColinPitrat/caprice32#55]
I tried compiling Caprice32 under MS Windows (Win8 laptop) yesterday night. The experience was far from pleasant :-(
I first tried and install the latest MSYS2 environment from http://www.msys2.org/, install all the relevant packages (gcc, SDL 1.2.15, freetype, zlib, libpng, git make), clone the the git repo and try a plain "make windows". Looks like the WINCXX variable is incorrect and msys2 uses g++ as a compiler. After fixing this, the  compilation starts then miserably fail because z80.c redefines _A, _B, etc. which seem to be used by C++ system libraries under MSYS (?).
I then used an old mingw + msys install I had on the same laptop, and could finally build a windows executable after pretty heavy changes to the makefile. My mingw  install by itself is a mess with dependencies installed all over the place, and a mix of libraries recompiled from source and binaries fetched from who knows where.
Do you have a working MS Windows compilation environment setup ? If yes could you share it (Cygwin? mingw? msys? msys2?).
We could move away the compilation instructions from the README.md to an INSTALL file - the way it is done in most projects - and make the INSTALL file more verbose about the environment ?

**Comment by ColinPitrat** [comment]
I don't have windows. I use `make windows` to cross compile from ArchLinux using mingw (hence the default WINCXX `i686-w64-mingw32-g++`). I think (but not 100% sure) the packages I had to install are:

- mingw-w64
- mingw-w64-sdl
- mingw-w64-zlib
- mingw-w64-freetype2-bootstrap
- mingw-w64-pkg-config

I'd really like to have a clean way to build and deliver from Windows. Unfortunately, it requires more than just building ... Last build I tested on windows had issues that would probably need long investigations to be fixed (e.g sound issues).
